File: Basics/Conditional-Statements-Advanced/Exercise/fishing_boat.py
budget = int(input())
season = input()
fishers_count = int(input())
boat_rent = 0

if season == "Spring":
    boat_rent = 3000
    if fishers_count <= 6:
        boat_rent -= boat_rent * 0.10
    elif 7 <= fishers_count <= 11:
        boat_rent -= boat_rent * 0.15
    elif fishers_count >= 12:
        boat_rent -= boat_rent * 0.25
elif season == "Summer" or season == "Autumn":
    boat_rent = 4200
    if fishers_count <= 6:
        boat_rent -= boat_rent * 0.10
    elif 7 <= fishers_count <= 11:
        boat_rent -= boat_rent * 0.15
    elif fishers_count >= 12:
        boat_rent -= boat_rent * 0.25
elif season == "Winter":
    boat_rent = 2600
    if fishers_count <= 6:
        boat_rent -= boat_rent * 0.10
    elif 7 <= fishers_count <= 11:
        boat_rent -= boat_rent * 0.15
    elif fishers_count >= 12:
        boat_rent -= boat_rent * 0.25

if not season == "Autumn":
    if fishers_count % 2 == 0:
        boat_rent -= boat_rent * 0.05

money_left_or_needed = abs(budget - boat_rent)

if budget >= boat_rent:
    print(f"Yes! You have {money_left_or_needed:.2f} leva left.")
else:
    print(f"Not enough money! You need {money_left_or_needed:.2f} leva.")

File: JS-Basics/0-3-Conditional-Steatements-Advanced/Exercise/fishing-boat.js
function isBudgetEnough(input) {
    let budget = Number(input[0]);
    let season = input[1];
    let fishermanCount = Number(input[2]);
    let price = 0;
    if (season == "Spring") {
        price = 3000;
        if (fishermanCount <= 6) {
            price -= price * 0.10;
        } else if (fishermanCount <= 11) {
            price -= price * 0.15;
        } else if (fishermanCount > 11) {
            price -= price * 0.25;
        }
    } else if (season == "Summer" || season == "Autumn") {
        price = 4200;
        if (fishermanCount <= 6) {
            price -= price * 0.10;
        } else if (fishermanCount <= 11) {
            price -= price * 0.15;
        } else if (fishermanCount > 11) {
            price -= price * 0.25;
        }
    } else if (season == "Winter") {
        price = 2600;
        if (fishermanCount <= 6) {
            price -= price * 0.10;
        } else if (fishermanCount <= 11) {
            price -= price * 0.15;
        } else if (fishermanCount > 11) {
            price -= price * 0.25;
        }
    }

    if (season != "Autumn" && fishermanCount % 2 == 0) {
        price -= price * 0.05;
    }
    
    let diff = Math.abs(price - budget);
    if (price <= budget) {
        console.log(`Yes! You have ${diff.toFixed(2)} leva left.`);
    } else {
        console.log(`Not enough money! You need ${diff.toFixed(2)} leva.`);
    }
}

isBudgetEnough(["2000", "Winter", "13"])
